awx/settings/defaults.py

```diff
@@ -853,6 +853,7 @@
         'awx.main.signals': {'level': 'INFO'},  # very verbose debug-level logs
         'awx.api.permissions': {'level': 'INFO'},  # very verbose debug-level logs
         'awx.analytics': {'handlers': ['external_logger'], 'level': 'INFO', 'propagate': False},
+        'awx.analytics.broadcast_websocket': {'handlers': ['console', 'file', 'wsbroadcast', 'external_logger'], 'level': 'INFO', 'propagate': False},
```

Comment:
Looks right to me.

Reply:
hmm... this isn't quite jiving for me.

https://github.com/ansible/awx/blob/ef29589940bb071b802671f090158f0916f9c260/awx/main/conf.py#L572

The external logging setting allows you to select which loggers to enable. So if you logged to a new `awx.analytics.____`, then yes, it makes sense that we would miss it. Unless we had `awx.analytics` enabled by default and the new loggers propagated (which they did before, but won't now).

I don't think we can figure this out completely without also looking at the dynamic logic there is for the external logger. But I admit I don't fully understand the fix.... although if you added it to the default loggers list, that would make sense to me (with the incompletely information I have now).

Reply:
This clears up the rest for me.

https://github.com/ansible/awx/blob/ef29589940bb071b802671f090158f0916f9c260/awx/main/utils/filters.py#L94-L100

A few takeaways:
 - your fix still appears _incomplete_, you added "broadcast_websocket" here, but it also needs to be listed among the default external loggers
 - anything sent to `awx.analytics` is just plain wrong. That implies that nothing should propagate down to that logger

This latter point is... oof. To prevent regressions of this type, we should enforce that nothing ever logs to an un-documented sub-logger of `awx.analytics`. Maybe that can be done with a unit test?

Reply:
Here we go, this can't be used to fix the problem, but it encapsulates the issue.

```
In [7]: [k for k in logging.Logger.manager.__dict__['loggerDict'].keys() if k.startswith('awx.analytics')]
Out[7]: 
['awx.analytics',
 'awx.analytics.performance',
 'awx.analytics.job_lifecycle',
 'awx.analytics.job_events',
 'awx.analytics.system_tracking',
 'awx.analytics.activity_stream']

In [8]: from awx.main.analytics import broadcast_websocket

In [9]: [k for k in logging.Logger.manager.__dict__['loggerDict'].keys() if k.startswith('awx.analytics')]
Out[9]: 
['awx.analytics',
 'awx.analytics.performance',
 'awx.analytics.job_lifecycle',
 'awx.analytics.job_events',
 'awx.analytics.system_tracking',
 'awx.analytics.activity_stream',
 'awx.analytics.broadcast_websocket']
```

We can tell via inspection of the global python objects which loggers have been created (at least on import... code can always follow non-standard practice). But that doesn't work unless everything in our app has been imported, which is hard to guarantee. So if you could do a hack job to accomplish that goal, fixing it would be possible by this approach.